src/test/java/bio/terra/workspace/service/resource/ValidationUtilsTest.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.workspace.service.resource;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.service.resource.referenced.exception.InvalidReferenceException;
+import org.junit.jupiter.api.Test;
+
+public class ValidationUtilsTest extends BaseUnitTest {
+  @Test
+  public void aiNotebookInstanceName() {
+    ValidationUtils.validateAiNotebookInstanceId("valid-instance-id-0");
+    assertThrows(
+        InvalidReferenceException.class,
+        () -> ValidationUtils.validateAiNotebookInstanceId("1number-first"));
+    assertThrows(
+        InvalidReferenceException.class,
+        () -> ValidationUtils.validateAiNotebookInstanceId("-dash-first"));
+    assertThrows(
+        InvalidReferenceException.class,
+        () -> ValidationUtils.validateAiNotebookInstanceId("no whitespace"));
```

Comment:
nit: all the other names are describing themselves, but this one should be something like `white space` so it's still positive.

Reply:
done